### PR TITLE
 Add backward simulation process

### PR DESCRIPTION
**Pull request by kryv**
- `Machine.propagate` support backward propagation.
- `State` object provide last transfer matrix information.

### CODE REVIEW

File: sphinx_doc/tutorial.rst [hunk 1]
```diff
@@ -105,15 +105,24 @@ Check new parameter of the solenoid.
 
 .. image:: figs/atob.png
 
-User can input *start-point index* and *end-point index* to :py:func:`propagate <Machine.propagate>`.
+User can input *start-point index* and *propagation number* to :py:func:`propagate <Machine.propagate>`.
 
->>> M.propagate(S, 0, 10) # simulate from 0th to the entrance of 10th element
+>>> M.propagate(S, 0, 10) # simulate from 0th to 9th element
```
Comment (zhangt58) [inline sphinx_doc/tutorial.rst:110]: Can I do `M.propagate(S, 10, 1)` to back-track from 10th to 1st element?

File: sphinx_doc/beam.rst [hunk 2]
```diff
@@ -127,6 +129,10 @@ State Class
 
         **float**: Driven phase of the last rf cavity. [deg]
 
+    .. py:attribute:: transfer
```
Comment (zhangt58) [inline sphinx_doc/beam.rst]: The new attribute name `transfer` is a verb, but this is should not be an action, can we name it with a noun, e.g. `m_transfer` or any other word which can properly describe this attribute?

File: sphinx_doc/tutorial.rst (hunk 1 above)
Comment (kryv) [reply sphinx_doc/tutorial.rst:110]: Possible, but it is inconsistent with the forward-track case. 
In FLAME, ``M.propagate(S, 10, 1)`` means forward-simulation of 10th element.

File: sphinx_doc/beam.rst (hunk 2 above)
Comment (kryv) [reply sphinx_doc/beam.rst]: I understand. I will change it to ``transmat``.

**Pull request merged**